pkg/graphql/federation/federation_config.go

```diff
@@ -0,0 +1,161 @@
+package federation
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/astparser"
+	graphqlDataSource "github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/graphql_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/httpclient"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+	"github.com/jensneuse/graphql-go-tools/pkg/federation"
+	"github.com/jensneuse/graphql-go-tools/pkg/graphql"
+)
+
+const (
+	keyDirectiveName      = "key"
+	requireDirectiveName  = "requires"
+	externalDirectiveName = "external"
+)
+
+func NewEngineConfigV2Factory(httpClient *http.Client, dataSourceConfig ...graphqlDataSource.Configuration) *EngineConfigV2Factory {
+	return &EngineConfigV2Factory{
+		httpClient:        httpClient,
+		dataSourceConfigs: dataSourceConfig,
+	}
+}
+
+type EngineConfigV2Factory struct {
+	httpClient        *http.Client
+	dataSourceConfigs []graphqlDataSource.Configuration
+	schema            *graphql.Schema
+}
+
+func (f *EngineConfigV2Factory) Schema() (*graphql.Schema, error) {
```

Comment:
I would call this function `MergedSchema()` as `Schema()` does not fully represent what is happening under the hood